app/jobs/insert_post_to_suffix_tree_job.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+class InsertPostToSuffixTreeJob < ApplicationJob
+  queue_as :default
+
+  def perform(post_id)
+    Rails.logger.info "Job started (#{DateTime.now}, +0s)"
+    SuffixTreeHelper.insert_post post_id
+    SuffixTreeHelper.sync_async
+    Rails.logger.info "Job finished (#{DateTime.now})"
+  end
+end
```

Comment:
You don't need this job, just alter the batch job to take args as a splat with `def perform(*post_ids)` and then `post_ids = post_ids.flatten` if you want to be extra safe but I don't think you need that.